src/util/string_utils.h

```diff
@@ -48,21 +48,35 @@ std::string trim_from_last_delimiter(
 /// \param e: Iterator pointing past last item to print
 /// \param delimiter: Object to print between each item in the iterator range
 /// \return A reference to the ostream that was passed in
-template<typename Stream, typename It, typename Delimiter>
+template <typename Stream, typename It, typename Delimiter>
+Stream &
+join_strings(Stream &&os, const It b, const It e, const Delimiter &delimiter)
+{
+  using value_type = decltype(*b);
+  // Call auxiliary function with identity function
+  return join_strings(
+    os, b, e, delimiter, [](const value_type &x) { return x; });
+}
+
+template <typename Stream,
+          typename It,
+          typename Delimiter,
+          typename ToStringFunc>
 Stream &join_strings(
-  Stream &os,
+  Stream &&os,
```

Comment:
What is the added value of passing this as an rvalue?

Reply:
It's a "universal" reference rather than rvalue; With this it can take a normal lvalue reference as well as an rvalue reference. This doesn't make a huge difference, but allows a pattern like `join_strings(std::ostringstream{},....).str()` (without having to declare a variable for the stream) as we have in the tests (passing in an lvalue will still work as before).

Reply:
Thanks a lot, there is so much left to learn...